### PR TITLE
Remove old DI container from various files

### PR DESCRIPTION
**Pull request by Tithugues**
### Description

Try to get rid of PhpMyAdmin\DI and replace it by Symfony DI.
Some tasks are needed, like giving all needed objects to every scripts / classes (instead of getting instance of container) and use `db`, `table`, … as DI parameters instead of updating the definition.

Related to #15266 and #15271

Before submitting pull request, please review the following checklist:

- [x] Make sure you have read our [CONTRIBUTING.md](https://github.com/phpmyadmin/phpmyadmin/blob/master/CONTRIBUTING.md) document.
- [x] Make sure you are making a pull request against the correct branch. For example, for bug fixes in a released version use the corresponding QA branch and for new features use the _master_ branch. If you have a doubt, you can ask as a comment in the bug report or on the mailing list.
- [ ] Every commit has proper `Signed-off-by` line as described in our [DCO](https://github.com/phpmyadmin/phpmyadmin/blob/master/DCO). This ensures that the work you're submitting is your own creation.
- [ ] Every commit has a descriptive commit message.
- [ ] Every commit is needed on its own, if you have just minor fixes to previous commits, you can squash them.
- [ ] Any new functionality is covered by tests.

### CODE REVIEW

**Comment by Tithugues** [comment]
@mauriciofauth , you asked if we'll be able to remove existing DI. "Winter is coming." I mean… "Removal is coming."
(There are a lot of commits here because there are also the commits of the other PR.)

File: libraries/classes/Di/Migration.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    public function __construct(ContainerBuilder $containerBuilder)
+    {
+        $this->containerBuilder = $containerBuilder;
+    }
+
+    /**
+     * Get the instance of the service
+     *
+     * @param string $key
+     * @param        $value
```
Comment (mauriciofauth) [inline libraries/classes/Di/Migration.php]: You need to follow the format `@param type $name Description` here.

File: libraries/classes/Di/Migration.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ */
+class Migration
+{
+    /** @var self */
+    protected static $instance;
+
+    /** @var ContainerBuilder */
+    protected $containerBuilder;
+
+    public static function getInstance(?ContainerBuilder $containerBuilder = null): self
```
Comment (mauriciofauth) [inline libraries/classes/Di/Migration.php]: You need to add the PHPDoc for this method.

File: libraries/classes/Di/Migration.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        if (null === $containerBuilder) {
+            throw new InvalidArgumentException('Container builder should be sent for ' . self::class . ' creation');
+        }
+
+        return self::$instance = new self($containerBuilder);
+    }
+
+    public function __construct(ContainerBuilder $containerBuilder)
```
Comment (mauriciofauth) [inline libraries/classes/Di/Migration.php]: You need to add the PHPDoc for the constructor.

File: db_central_columns.php [hunk 4]
```diff
@@ -14,41 +14,16 @@
 use PhpMyAdmin\CentralColumns;
 use PhpMyAdmin\Controllers\Database\CentralColumnsController;
 use PhpMyAdmin\Core;
-use PhpMyAdmin\DatabaseInterface;
-use PhpMyAdmin\Di\Container;
 use PhpMyAdmin\Message;
 use PhpMyAdmin\Response;
-use Symfony\Component\DependencyInjection\Definition;
-
-global $db;
```
Comment (mauriciofauth) [inline db_central_columns.php]: Removing `global $db;` will cause PHPStan to fail.

File: db_events.php [hunk 5]
```diff
@@ -9,35 +9,24 @@
...
 use PhpMyAdmin\Response;
 use PhpMyAdmin\Url;
 use PhpMyAdmin\Util;
-use Symfony\Component\DependencyInjection\Definition;
 
 if (! defined('ROOT_PATH')) {
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
 
-global $db, $table;
```
Comment (mauriciofauth) [inline db_events.php]: Removing `global $db, $table;` will cause PHPStan to fail.

File: db_routines.php [hunk 6]
```diff
@@ -10,38 +10,27 @@
...
 use PhpMyAdmin\Response;
 use PhpMyAdmin\Url;
 use PhpMyAdmin\Util;
-use Symfony\Component\DependencyInjection\Definition;
 
 if (! defined('ROOT_PATH')) {
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
 
-global $db, $table;
```
Comment (mauriciofauth) [inline db_routines.php]: Removing `global $db, $table;` will cause PHPStan to fail.

File: db_triggers.php [hunk 7]
```diff
@@ -9,35 +9,24 @@
...
 use PhpMyAdmin\Response;
 use PhpMyAdmin\Url;
 use PhpMyAdmin\Util;
-use Symfony\Component\DependencyInjection\Definition;
 
 if (! defined('ROOT_PATH')) {
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
 
-global $db, $table;
```
Comment (mauriciofauth) [inline db_triggers.php]: Removing `global $db, $table;` will cause PHPStan to fail.

File: tbl_addfield.php [hunk 8]
```diff
@@ -20,18 +19,13 @@
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
 
-global $cfg, $db, $table;
```
Comment (mauriciofauth) [inline tbl_addfield.php]: Removing `global $cfg, $db, $table;` will cause PHPStan to fail.

File: tbl_create.php [hunk 9]
```diff
@@ -20,23 +19,19 @@
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
 
-global $cfg, $db, $table;
```
Comment (mauriciofauth) [inline tbl_create.php]: Removing `global $cfg, $db, $table;` will cause PHPStan to fail.

File: tbl_find_replace.php [hunk 10]
```diff
@@ -10,27 +10,17 @@
...
 use PhpMyAdmin\Controllers\Table\SearchController;
-use PhpMyAdmin\Di\Container;
-use PhpMyAdmin\Response;
 use Symfony\Component\DependencyInjection\Definition;
 
 if (! defined('ROOT_PATH')) {
     define('ROOT_PATH', __DIR__ . DIRECTORY_SEPARATOR);
 }
 
-global $url_query;
```
Comment (mauriciofauth) [inline tbl_find_replace.php]: Removing `global $url_query;` will cause PHPStan to fail.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

File: libraries/classes/Di/Migration.php (hunk 2 above)
Comment (mauriciofauth) [reply libraries/classes/Di/Migration.php]: This will cause phpcs to fail.

File: libraries/classes/Di/Migration.php (hunk 3 above)
Comment (mauriciofauth) [reply libraries/classes/Di/Migration.php]: This will cause phpcs to fail.

File: libraries/classes/Di/Migration.php (hunk 1 above)
Comment (mauriciofauth) [reply libraries/classes/Di/Migration.php]: This will cause phpcs and phpstan to fail.

**Comment by mauriciofauth** [comment]
Hi, @Tithugues, thank you for your work on this pull request.

How much do you think it takes before we can remove our custom container?
We are wanting to feature freeze the version 5.0 and we would like to include this pull request.

**Comment by Tithugues** [comment]
@mauriciofauth ,
Good question… I don't really know how long it could be. Maybe 2-3 weeks. I hope not longer.
When do you expect to create the QA_5_0 branch please?

**Review by MauricioFauth — APPROVED** [review]

**Comment by mauriciofauth** [comment]
Hi, @Tithugues,

I merged this pull request to include this changes in 5.0. We are planning to release 5.0.0-alpha soon, so the remaining changes may not be included.
You can open a new pull request with the remaining changes.

Thank you for your contribution.

**Comment by ibennetch** [comment]
I think we are waiting on this to release 5.0.0-alpha. I’m willing to wait
about two weeks, so long as no one else objects. I think that is worth it
to get this in to 5.0.

**Comment by mauriciofauth** [comment]
I open #15335 to help with the removal of the container.

**Pull request merged**